guides/source/error_reporting.md

```diff
@@ -117,11 +126,32 @@ rescue StandardError => e
 end
 ```
 
-Any options you pass will be passed on the error subscribers.
+Any options you pass will be passed on to the error subscribers.
+
+#### Reporting Unexpected Errors
+
+You can report any errors that may unexpectedly occur when the imagined conditions of your code are not met by calling [`Rails.error.unexpected`](https://edgeapi.rubyonrails.org/classes/ActiveSupport/ErrorReporter.html#method-i-unexpected").
+
+When called in production, this method will return nil after the error is reported and the execution of your code will continue.
+
+When called in development, the error will be wrapped in a new error class (to ensure it's not being rescued higher in the stack) and surfaced to the developer for debugging.
+
+For example:
+
+```ruby
+def edit
+  if published?
+    Rails.error.unexpected("[BUG] Attempting to edit a published article, that shouldn't be possible")
+    return false
+  end
+  # ...
+end
+```
+NOTE: This method is intended to gracefully handle any errors that may occur in production, but that aren't anticipated to be the result of typical use.
```

Comment:
Looks like a newline is missing after the code fence, rubocop will complain about this.